### DISCUSSION TITLE
Multiple assignment with tar_target

### DISCUSSION THREAD

**Discussion by cbellard** [ropensci/targets#733] [Help]
I would like to know if multiple assignment is possible with targets similarly to zeallot package, i.e.:
```
c(a,b) %<-% c(1,2)
tar_target(c(data_train,data_test),list(data[ind_train,],data[ind_test,]))
```

It would improve the readability of the code I think.

**Comment by wlandau** [comment]
Sorry, I do not have plans to support multiple assignment in `targets`. But [dynamic branching](http://books.ropensci.org/targets/dynamic.html) has similar properties.

**Reply by cbellard** [reply]
Thanks a lot. I am very new to this coding way and I did not manage to use dynamic branching for my problem. Here is my attempt (model is created but predict does not work) :

```library(targets)

list(
  tar_target(ind,list(ind_train=1:3,ind_test=5:7)),
  tar_target(data,data.frame(x=1:9,y=1:9+runif(9))),
  tar_target(model, lm(y~x,data[unlist(ind),]), pattern = slice(ind,1)),
  tar_target(y,predict(model,data.frame(x=data[unlist(ind),"x"])), pattern = slice(ind,2))
)
```

**Reply by wlandau** [reply]
Here's a version that does work. I used `iteration = "list"` for the index and model, and I renamed the `y` target to `y_predict`.

```r
library(targets)
list(
  tar_target(ind, list(ind_train = 1:3, ind_test = 5:7), iteration = "list"),
  tar_target(data, data.frame(x = 1:9, y = 1:9 + runif(9))),
  tar_target(model, lm(y ~ x, data[ind,]), pattern = slice(ind, 1), iteration = "list"),
  tar_target(y_predict, predict(model[[1]], data.frame(x = data[ind, "x"])), pattern = slice(ind, 2))
)
```

If you're interested in cleaner syntax for this specific example use case, you might also have a look at `tarchetypes::tar_plan()`.

```r
library(targets)
library(tarchetypes)
tar_plan(
  ind = list(ind_train = 1:3, ind_test = 5:7),
  data = data.frame(x = 1:9, y = 1:9 + runif(9)),
  model = lm(y ~ x, data[ind$ind_train, ]),
  y_predict = predict(model, data.frame(x = data[ind$ind_test, "x"]))
)
```

**Reply by cbellard** [reply]
Thanks for the help !
Based on your answer I found  an alternative way using zeallot multiple assignment function.
```
library(targets)
library(tarchetypes)
library(zeallot)
c(ind_train,ind_test) %<-% list(1:4,5:8)
tar_plan(
  data = data.frame(x = 1:9, y = 1:9 + runif(9)),
  model = lm(y ~ x, data[ind_train, ]),
  y_predict = predict(model, data.frame(x = data[ind_test, "x"]))
)

```
